lib/common/imageUtil.py

```diff
@@ -0,0 +1,359 @@
+                    src_img = Image.open(input_image_data['input_fp'])
+                    dst_img = src_img.crop(crop_region)
+                    dst_img.save(input_image_data['output_fp'])
+                else:
+                    logger.warning("Incorrect image format of file during crop images: [%s]" % input_image_data['input_fp'])
+                    continue
+        except Exception as e:
+            logger.error(e)
+
+
+def compare_with_sample_image_multi_process(input_sample_data, input_image_data, input_settings):
+    """
+    Compare sample images, return matching list.
+    @param input_sample_dp: input sample folder path
+    @return: the matching result list
+    """
+    manager = Manager()
+    result_list = manager.list()
+    p_list = []
+    for search_direction in input_settings['event_points'].keys():
+        parallel_run_settings = copy.deepcopy(input_settings)
+        parallel_run_settings['search_direction'] = search_direction
+        args = [input_sample_data, input_image_data, parallel_run_settings, result_list]
+        p_list.append(Process(target=parallel_compare_image, args=args))
+        p_list[-1].start()
+    for p in p_list:
+        p.join()
+    map_result_list = sorted(map(dict, result_list), key=lambda k: k['time_seq'])
+    logger.info(map_result_list)
+    return map_result_list
+
+
+def get_search_range(input_time_stamp, input_fps, input_image_list_len=0, input_cover_sec=10):
+    ref_start_point = input_time_stamp[1] - input_time_stamp[0]
+    ref_end_point = input_time_stamp[2] - input_time_stamp[0]
+    if input_image_list_len:
+        search_range = [
+            # For finding the beginning, the range cannot start less than zero.
+            max(int((ref_start_point - input_cover_sec) * input_fps), 0),
+            min(int((ref_start_point + input_cover_sec) * input_fps), input_image_list_len - 1 ),
+            # For finding the end, the range cannot start less than zero.
+            max(int((ref_end_point - input_cover_sec) * input_fps), 0),
+            min(int((ref_end_point + input_cover_sec) * input_fps), input_image_list_len - 1)]
+    else:
+        search_range = [
+            # For finding the beginning, the range cannot start less than zero.
+            max(int((ref_start_point - input_cover_sec) * input_fps), 0),
+            int((ref_start_point + input_cover_sec) * input_fps),
+            # For finding the end, the range cannot start less than zero.
+            max(int((ref_end_point - input_cover_sec) * input_fps), 0),
+            int((ref_end_point + input_cover_sec) * input_fps)]
+    logger.info("Image Comparison search range [%s] when image list len is [%s]: " % (str(search_range), str(input_image_list_len)))
+    return search_range
+
+
+def parallel_compare_image(input_sample_data, input_image_data, input_settings, result_list):
+    # init value
+    sample_dct = None
+    image_fn_list = copy.deepcopy(input_image_data.keys())
+    image_fn_list.sort()
```

Comment:
sort file name with number need to consider string type of number
or the will be [img_1.bmp, img_1x.bmp, img_1xxx.bmp, img_2.bmp, img_2x.bmp, img_2xxx.bmp, etc.]
`from commonUtil import CommonUtil`
`image_fn_list.sort()` -> `image_fn_list.sort(key=CommonUtil.natural_keys)`